wandb/board/ui/src/util/apollo.js

```diff
@@ -110,25 +116,37 @@ const errorLink = onError(({networkError, graphQLErrors}, store) => {
         }
         dispatch(push('/login'));
       } else {
-        console.error(`GraphQL error ${message} (${code}):`);
         dispatch(displayError(error));
       }
     });
+    dispatch(setFlash({message: errorMessage, color: 'red'}));
   }
   if (networkError) {
-    console.error(`Network Error: ${networkError}`);
     if (networkError.result) {
       console.error(networkError.result.errors);
+    } else if (networkError.message === 'Failed to fetch') {
+      errorMessage = 'Network Error';
+      dispatch(updateGraphqlStatus(false));
     }
 
-    dispatch(setFlash({message: 'Backend Unavailable', color: 'red'}));
+    if (networkError.statusCode === 400) {
+      dispatch(
+        displayError({
+          code: networkError.statusCode,
+          message: errorMessage,
+        })
+      );
+    } else {
+      dispatch(setFlash({message: errorMessage, color: 'red'}));
```

Comment:
It feels weird to me that we flash here and on the error page, but on the error page we'll refresh the content.  Why not dispatch `displayError` here?

Reply:
This flash message will be triggered for Network error and we don't need error page for that.

Reply:
It won't, not without changing ErrorPage